hikey.mk

```diff
@@ -123,26 +138,33 @@ edk2-clean: edk2-clean-common
 ################################################################################
 LINUX_DEFCONFIG_COMMON_ARCH ?= arm64
 LINUX_DEFCONFIG_COMMON_FILES ?= $(LINUX_PATH)/arch/arm64/configs/defconfig \
-				$(CURDIR)/kconfigs/hikey.conf
+				$(CURDIR)/kconfigs/hikey.conf \
+				$(PATCHES_PATH)/kernel_config/usb_net_dm9601.conf \
+				$(PATCHES_PATH)/kernel_config/ftrace.conf
 
 linux-defconfig: $(LINUX_PATH)/.config
 
 linux-gen_init_cpio: linux-defconfig
-	make -C $(LINUX_PATH)/usr \
+	$(MAKE) -C $(LINUX_PATH)/usr \
 		CROSS_COMPILE=$(CROSS_COMPILE_NS_KERNEL) \
 		ARCH=arm64 \
 		LOCALVERSION= \
 		gen_init_cpio
 
-LINUX_COMMON_FLAGS += ARCH=arm64 Image modules dtbs
+LINUX_COMMON_FLAGS += ARCH=arm64 Image modules
+UPSTREAM_KERNEL = 0
+ifeq ($(UPSTREAM_KERNEL),0)
+LINUX_COMMON_FLAGS += hi6220-hikey.dtb
+DTB = $(LINUX_PATH)/arch/arm64/boot/dts/hi6220-hikey.dtb
+else
+LINUX_COMMON_FLAGS += hisilicon/hi6220-hikey.dtb
+DTB = $(LINUX_PATH)/arch/arm64/boot/dts/hisilicon/hi6220-hikey.dtb
+endif
```

Comment:
That's OK but this would be even better:
```
UPSTREAM_KERNEL := $(if $(wildcard $(LINUX_PATH)/arch/arm64/boot/dts/hisilicon/hi6220-hikey.dts),1,0)
```